public_html/wp-content/themes/pattern-directory/src/components/copy-pattern-button/index.js

```diff
@@ -1,35 +1,71 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
 /**
  * External dependencies
  */
 import { __ } from '@wordpress/i18n';
+import { speak } from '@wordpress/a11y';
 import { Button } from '@wordpress/components';
+import { useEffect, useState } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import { copyToClipboard } from '../../utils';
 
-const CopyPatternButton = ( { onSuccess } ) => {
-	const handleClick = ( { target } ) => {
+const CopyPatternButton = ( { isSmall = false, onSuccess, content } ) => {
+	const [ copied, setCopied ] = useState( false );
+
+	if ( ! content ) {
 		// Grab the pattern markup from hidden input
 		const blockData = document.getElementById( 'block-data' );
-		const blockPattern = JSON.parse( decodeURIComponent( blockData.value ) );
+		content = JSON.parse( decodeURIComponent( blockData.value ) );
+	}
+
+	const handleClick = ( { target } ) => {
+		const success = copyToClipboard( content );
 
-		const success = copyToClipboard( blockPattern );
+		setCopied( success );
 
 		// Make sure we reset focus in case it was lost in the 'copy' command.
 		target.focus();
 
-		if ( success ) {
+		if ( success && 'function' === typeof onSuccess ) {
```

Comment:
Would it make sense to set a default value for `onSuccess` to communicate its optionality as opposed to check it as a function?  When I see this I always think.. "What else could it be?"

Reply:
Well, it could be undefined. I don't think a default value makes sense, because we don't _need_ to do anything here.

Reply:
Gutenberg controls typically use `lodash.noop`:

https://github.com/WordPress/gutenberg/blob/cdbab7fff0d285787a565fe0e23b620b28ab1922/packages/components/src/input-control/index.js#L37